go/vt/vttablet/tabletmanager/tm_state.go

```diff
@@ -229,12 +229,21 @@ func (ts *tmState) updateLocked(ctx context.Context) {
 		return
 	}
 
+	populateMetadata := func() {
```

Comment:
Populate metadata does some extra work that only makes sense when the tablet is bootstrapping (like creating the metadata table). All the instructions are idempotent, but I think it will be clearer/cleaner to explicitly add a method to  only update the metadata